src/zope/app/component/testing.py

```diff
@@ -11,185 +11,44 @@
 # FOR A PARTICULAR PURPOSE.
 #
 ##############################################################################
-"""Base Mix-in class for Placeful Setups
-
-Also contains common test related classes/functions/objects.
 """
+Test helpers.
+"""
+from __future__ import absolute_import
 
-import os
 import zope.interface
 import zope.site.folder
-from zope.component.interfaces import IComponentLookup
-from zope.app.component.interfaces import ILocalSiteManager
-from zope.app.testing import setup
-from zope.app.testing.placelesssetup import PlacelessSetup
-
-from zope.app.testing.functional import ZCMLLayer
-from zope.traversing.api import traverse
-
-AppComponentLayer = ZCMLLayer(
-    os.path.join(os.path.split(__file__)[0], 'ftesting.zcml'),
-    __name__, 'AppComponentLayer', allow_teardown=True)
-
-class Place(object):
-
-    def __init__(self, path):
-        self.path = path
-
-    def __get__(self, inst, cls=None):
-        if inst is None:
-            return self
-
-        try:
-            # Use __dict__ directly to avoid infinite recursion
-            root = inst.__dict__['rootFolder']
-        except KeyError:
-            root = inst.rootFolder = setup.buildSampleFolderTree()
-
-        return traverse(root, self.path)
-
-
-class PlacefulSetup(PlacelessSetup):
```

Comment:
zope.app.container has already had this dependency expunged. I expect to get to zope.app.applicationcontrol and zope.app.authentication. I was hoping to ultimately remove that dependency, as I did in [zope.app.rotterdam](https://github.com/zopefoundation/zope.app.rotterdam/pull/1/files#diff-538080e0a96b8a86c9a87a532855f6e7R64) by copying just the relevant portions. It was especially necessary there because they had a circular dependency, but I don't think the others do.

Many, but not all, of the contents that were removed could be added back without introducing a zope.app.testing dependency (as zope.app.rotterdam demonstrates).  Those other projects will likely need other changes to move on to Python 3, including dropping zope.app.testing, so perhaps preserving this here isn't that useful? Or maybe the de-duplication is worth it if it substantially eases porting.